website/blog/2020-06-25-wavemaker-openapi-postgrest-support.md

```diff
@@ -0,0 +1,134 @@
+---
+title: "PostgREST OpenAPI support in WaveMaker"
+author: Nikhilesh K V
+---
+
+WaveMaker has added support for integration with [OpenAPIs](https://www.wavemaker.com/learn/blog/2020/04/21/wavemaker-openapi-import) in 10.4.  As we are trying out more and more variations of OpenAPI documents from different sources of generation tools, our OpenAPI parser is evolving too. One such example is when we tested an OpenAPI document generated through [PostgREST](http://postgrest.org/en/v7.0.0/). In this blog we'll discuss what changes we have made in 10.5 in order to support OpenAPI documents generated from such sources.
+<!-- truncate -->
+
+![OpenApi](/learn/assets/postGRESTlogo.png)
+
+## Introduction to PostgREST
+
+PostgREST is a standalone web server that turns your PostgreSQL database directly into a RESTful API. The structural constraints and permissions in the database determine the API endpoints and operations.
+
+## Structure of PostgREST generated OpenAPI Documents
+
+Generally, the structure of an OpenAPI document generated out of a Spring/Node etc app looks like this.
+```
+   {
+       "paths":{
+           "/employees":{
+               "post":{
+                   "parameters":[
+                       {
+                           "in":"body",
+                           "name":"employee",
+                           "description":"employee",
+                           "required":true,
+                           "schema":{
+                                "$ref":"#/definitions/Employee"
+                           }
+                       }
+                   ]
+
+               }
+           }
+       },
+       "definitions":{
+           "Employee":{
+               "type":"object",
+               "properties":{
+                   "empId":{
+                        "type":"string"
+                   },
+                   "empName":{
+                        "type":"string"
+                   }
+               },
+               "title":"Employee"
+           }
+       }
+   }
+```
+The above document indicates that the POST API expects a body parameter of the type Employee, therefore expecting empId and empName.
+
+Now let us see an OpenAPI document generated out of PostgREST.
+```
+{
+    "paths":{
+        "/todos":{
+            "post":{
+                "parameters":[
+                    {
+                        "$ref":"#/parameters/body.todos"
```

Comment:
We need to add what happened in 10.4 because we didn't recognize these parameters. Users were able to import the API but could not build form with right set of fields?